### PR TITLE
fix #2531: make WHERE with multiple OR terms work

### LINKED ISSUE #2531: More than two OR-ed expressions in SELECT not properly evaluated
In the following example the second response is obviously wrong.
It makes no difference if the field named 'type' is a tag or just a field.
If the query is changed to span multiple series, this can lead to errors like #2299 if the `value` field has different types and a function like `mean()` is applied.

```sql
> SELECT type FROM "my.pretty.series" WHERE type = 'data1c' OR type = 'bit'
name: my.pretty.series
-----------------------------------------------------------
time    type
> SELECT type FROM "my.pretty.series" WHERE type = 'data1c' OR type = 'bit' OR type = 'data1b'
name: my.pretty.series
-----------------------------------------------------------
time                    type
2015-05-10T17:22:25Z    byteEnum
2015-05-10T17:22:28Z    byteEnum
2015-05-10T17:27:27Z    byteEnum
2015-05-10T17:27:35Z    byteEnum
2015-05-10T17:27:55Z    byteEnum
2015-05-10T17:28:05Z    byteEnum
2015-05-10T17:28:25Z    byteEnum
```

I'm running v0.9.0-rc29 with:
TOML_GH_COMMIT=         056c9bc # github.com/BurntSushi/toml
PAT_GH_COMMIT=          b8a3500 # github.com/bmizerany/pat
BOLT_GH_COMMIT=         2c04100 # github.com/boltdb/bolt
GOLLECTD_GH_COMMIT=     cf6dec9 # github.com/kimor79/gollectd
LINER_GH_COMMIT=        1bb0d1c # github.com/peterh/liner
STATIK_GH_COMMIT=       274df12 # github.com/rakyll/statik
GOCRYPTO_GH_COMMIT=     24ffb5f # golang.org/x/crypto

with `go1.4.2 freebsd/amd64`on 10.1-RELEASE. I'm using it from a non-official freebsd-port. To have it reproducible the commit-id's from the dependencies are hardcoded in the recipe. It would be nice if influxdb defines them (maybe along with using `godep`).

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

**Comment by pauldix** [comment]
The filters stuff is quite tangled so I do have one question, which may be in the code but isn't totally obvious.

For filters where it's just a `true` filter, do you remove that from the list? I ask because the engine evaluates against any filter that's there and if you have a bunch of `true` filters that don't really need to exist, it'll slow performance.

And that's the most common case since you've made the `time` part of the where clause return all series with a true filter.

Seems like this could cause problems, which is why I was originally just ignoring the time part of the where clause when walking back up the parse tree to put together the filter expressions.

**Comment by dgnorton** [comment]
@pauldix made a change to convert `true` to `nil`

File: cmd/influxd/backup.go [hunk 1]
```diff
@@ -123,6 +123,8 @@ func (cmd *BackupCommand) nextPath(path string) (string, error) {
 			return "", err
 		}
 	}
+
+	return "", nil
```
Comment (toddboom) [inline cmd/influxd/backup.go]: it seems like this is throwing a build error when we run `go vet`:
```
cmd/influxd/backup.go:127: unreachable code
```

**Comment by dgnorton** [comment]
@toddboom oops, didn't mean to commit that line.  Made it so `godebug` would run but didn't mean to commit it.  Fixed.

**Comment by toddboom** [comment]
+1 with a rebase! ;)

**Pull request merged**